### ISSUE TITLE
Fix accessibility prop name inconsistencies

### ISSUE THREAD

**Issue by redallen** [patternfly/patternfly-react#1529]
Property names for [aria attributes](https://a11yproject.com/posts/ARIA-properties) are inconsistent. Some examples:
- `AboutModalContainer` takes the following properties: `ariaLabelledbyId`, `ariaDescribedById`. Note inconsistencies in `by` vs `By`.
- `AboutModalBox` takes the raw HTML properties: `aria-labelledby`, `aria-describedby`
- `Alert` takes `variant` and `variantLabel` and sets `aria-label=undefined` as a default property without it being an actual component property that we can typecheck.
- I'm sure there's more...

Codebase names:
- `ariaLabel`: 148
- `aria-label`: 424
- `aria-controls`: 10
- `ariaControls`: 3
- `aria-labelledby`: 74
- `ariaLabelledbyId`: 10
- `aria-describedby`: 18
- `ariaDescribedById`: 12
- `variant`: 694
- `variantLabel`: 13
- There's more... See the [aria property list](https://a11yproject.com/posts/ARIA-properties) and use a global regex tool to spot our naming inconsistencies.

We are already mostly using the HTML properties, so I propose we introduce a breaking change to rename properties like `ariaLabelledbyId` to their HTML counterpart `aria-labelledby`. If we need to do logic with them in the component, we can rename them to camel-case in the property destructuring that already takes place using the `:` syntax:
```javascript
{
  'variant-label': variantLabel,
  'aria-label': ariaLabel,
  'aria-labelledby': ariaLabeledBy,
  'aria-describedby': ariaDescribedBy
}
```

**Comment by redallen** [comment]
Ref #1525

**Comment by tlabaj** [comment]
closed by #3924

**Comment by tlabaj** [comment]
closed by #3924
